clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -679,4 +573,130 @@ public void patch(
       }
     }
   }
+
+  /**
+   * Handles validating whether kubernetes kinds should be cached by the current account, and
+   * whether metrics are enabled for the account.
+   */
+  private class KindValidator {
+    private final Supplier<String> checkNamespace = Suppliers.memoize(this::computeCheckNamespace);
+    private final Map<KubernetesKind, Boolean> readableKinds = new ConcurrentHashMap<>();
+    private final Supplier<Boolean> metricsReadable = Suppliers.memoize(this::checkMetricsReadable);
+
+    private final Set<KubernetesKind> kinds;
+    private final Set<KubernetesKind> omitKinds;
+
+    KindValidator(Set<KubernetesKind> kinds, Set<KubernetesKind> omitKinds) {
+      this.kinds = kinds;
+      this.omitKinds = omitKinds;
+    }
+
+    private String getCheckNamespace() {
+      return checkNamespace.get();
+    }
+
+    private String computeCheckNamespace() {
+      List<String> namespaces = getDeclaredNamespaces();
+
+      if (namespaces.isEmpty()) {
+        log.warn(
```

Comment:
use `{}` instead of String concatenation for log messages? It seems unlikely `warn` messages would be disabled, but still, good habit...